stl/inc/shared_mutex

```diff
@@ -220,19 +214,19 @@ public:
     shared_timed_mutex& operator=(const shared_timed_mutex&) = delete;
 
 private:
-    mutex _Mymtx;
-    condition_variable _Read_queue;
-    condition_variable _Write_queue;
-    _Read_cnt_t _Readers;
-    bool _Writing;
+    mutex _Mymtx{};
+    condition_variable _Read_queue{};
+    condition_variable _Write_queue{};
+    _Read_cnt_t _Readers = 0;
+    bool _Writing        = false;
 };
 
 _EXPORT_STD template <class _Mutex>
 class shared_lock { // shareable lock
 public:
     using mutex_type = _Mutex;
 
-    shared_lock() noexcept : _Pmtx(nullptr), _Owns(false) {}
+    shared_lock() = default;

```

Comment:
We have (or had?) a convention that we explicitly annotate defaulted functions with `noexcept` when the Standard does so even when the implicit _noexcept-specifier_ would be the same. It makes it easier for readers to correlate what they see with the specification.